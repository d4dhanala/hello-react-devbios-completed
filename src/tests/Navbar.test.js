import React from 'react';
import { render } from '@testing-library/react';
import Navbar from '../components/Navbar';
import renderer from 'react-test-renderer';
import {
    BrowserRouter as Router
  } from 'react-router-dom';


test('renders Home link', () => {
    const { getByText } = render(<Router ><Navbar /></Router>)
    const linkElement = getByText(/Home/i);
    expect(linkElement).toBeInTheDocument();
});

test('renders Developer Bios link', () => {
    const { getByText } = render(<Router ><Navbar /></Router>);
    const linkElement = getByText(/Developer Bios/);
    expect(linkElement).toBeInTheDocument();
});

test('renders Create Bios link', () => {
    const { getByText } = render(<Router ><Navbar /></Router>);
    const linkElement = getByText(/Create Bios/);
    expect(linkElement).toBeInTheDocument();
});

test('Home Navigates to proper url', () => {
    const { getByText } = render(<Router ><Navbar /></Router>);
    const linkElement = getByText(/Create Bios/);
    expect(linkElement).toBeInTheDocument();
});

test('Home navigates to correct route', ()=>{
    const component = renderer.create(<Router><Navbar /></Router>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    console.log(tree);
    let navlinks = tree.children[1].children[0]
    expect(navlinks.children[0].props.href).toBe("/");
    
})