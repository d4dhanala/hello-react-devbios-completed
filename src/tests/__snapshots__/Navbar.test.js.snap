// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home navigate to correct route 1`] = `
<nav
  className="navbar navbar-expand-sm bg-dark navbar-dark justify-content-between"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="logo"
      className="brand"
      src="logo.svg"
    />
  </a>
  <ul
    className="navbar-nav"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/bios"
        onClick={[Function]}
      >
        Developer Bios
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/create-bio"
        onClick={[Function]}
      >
        Create Bios
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Home navigates to correct route 1`] = `
<nav
  className="navbar navbar-expand-sm bg-dark navbar-dark justify-content-between"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="logo"
      className="brand"
      src="logo.svg"
    />
  </a>
  <ul
    className="navbar-nav"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/bios"
        onClick={[Function]}
      >
        Developer Bios
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/create-bio"
        onClick={[Function]}
      >
        Create Bios
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </li>
  </ul>
</nav>
`;

exports[`links navigate to correct route 1`] = `
<nav
  className="navbar navbar-expand-sm bg-dark navbar-dark justify-content-between"
>
  <a
    className="navbar-brand"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="logo"
      className="brand"
      src="logo.svg"
    />
  </a>
  <ul
    className="navbar-nav"
  >
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/bios"
        onClick={[Function]}
      >
        Developer Bios
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/create-bio"
        onClick={[Function]}
      >
        Create Bios
      </a>
    </li>
    <li
      className="nav-item"
    >
      <a
        className="nav-link"
        href="/search"
        onClick={[Function]}
      >
        Search
      </a>
    </li>
  </ul>
</nav>
`;
